packages/firebase_storage/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FlutterFirebaseStoragePlugin.java

```diff
@@ -202,255 +207,329 @@ private StorageReference getReference(Map<String, Object> arguments) {
   }
 
   private Task<Void> useEmulator(Map<String, Object> arguments) {
-    return Tasks.call(
-        cachedThreadPool,
-        () -> {
-          FirebaseStorage firebaseStorage = getStorage(arguments);
-          String host = (String) arguments.get("host");
-          int port = (int) arguments.get("port");
-          firebaseStorage.useEmulator(host, port);
-          return null;
-        });
+    TaskCompletionSource<Void> taskCompletionSource = new TaskCompletionSource<>();
+
+    cachedThreadPool.execute(() -> {
+      FirebaseStorage firebaseStorage = getStorage(arguments);
+      String host = (String) Objects.requireNonNull(arguments.get("host"));
+      int port = (int) Objects.requireNonNull((arguments.get("port")));
+      firebaseStorage.useEmulator(host, port);
```

Comment:
```suggestion
      firebaseStorage.useEmulator(host, port);
      taskCompletionSource.setResult(null);
```

this is missing I think